### PR TITLE
Fixing #11

### LINKED ISSUE #11: DownloadFiles - Allow accessing shared drive files

### PR DESCRIPTION
**Pull request by jefim**
Fixing issue #11

## Summary by CodeRabbit

- **New Features**
	- Added support for including shared drives in file searches.
	
- **Bug Fixes**
	- Improved handling of different file types and sizes in file searches.

- **Tests**
	- Added unit tests to verify the download functionality from shared drives.

- **Chores**
	- Updated version number to 1.1.0.

### CODE REVIEW

File: Frends.GoogleDrive.Download/Frends.GoogleDrive.Download/Frends.GoogleDrive.Download.cs [hunk 1]
```diff
@@ -1,143 +1,153 @@
...
+            service = await CreateDriveService(input, cancellationToken);
+            var files = GetFileList(service, input);
+            var success = true;
+
+            if (files != null && files.Count > 0)
+            {
+                foreach (var file in files)
+                {
+                    // Skip over folders
+                    if (file.Kind.Equals("drive#file") && file.Size is 0 or null)
```
Comment (MatteoDelOmbra) [inline Frends.GoogleDrive.Download/Frends.GoogleDrive.Download/Frends.GoogleDrive.Download.cs]: Won't it also skip empty files?

File: Frends.GoogleDrive.Download/Frends.GoogleDrive.Download/Frends.GoogleDrive.Download.cs [hunk 2]
```diff
@@ -1,143 +1,153 @@
...
+        fileListRequest.Fields = "nextPageToken, files(id, kind, name, size, version, createdTime)";
+        fileListRequest.Q = input.FileQuery;
+        fileListRequest.PageSize = 1000;
+
+        // shared drives support
+        fileListRequest.SupportsAllDrives = input.IncludeSharedDrives;
+        fileListRequest.IncludeItemsFromAllDrives = input.IncludeSharedDrives;
+
+        var ret = fileListRequest.Execute();
+        return ret.Files;
```
Comment (MatteoDelOmbra) [inline Frends.GoogleDrive.Download/Frends.GoogleDrive.Download/Frends.GoogleDrive.Download.cs:151]: Can be converted to async method

```suggestion
        var ret = await fileListRequest.ExecuteAsync(token);
```

File: Frends.GoogleDrive.Download/Frends.GoogleDrive.Download.Test/UnitTests.cs [hunk 3]
```diff
@@ -35,6 +35,29 @@ public void Setup()
         };
     }
 
+    [TestMethod]
+    [Ignore("Local testing only, requires shared drive setup.")]
```
Comment (MatteoDelOmbra) [inline Frends.GoogleDrive.Download/Frends.GoogleDrive.Download.Test/UnitTests.cs:39]: Why we can't test in on github? Can't we create some testing drive that will have some foobar file to download?
And if we want to ignore it on github, maybe it's worh to make it conditional? E.g. check if env.local exists
  [Here's](https://stackoverflow.com/questions/1594293/ignore-test-or-testfixture-based-on-a-condition) other example how we could implement conditional ignore.

**Review by MatteoDelOmbra — CHANGES_REQUESTED** [review]
Few comments added

File: Frends.GoogleDrive.Download/Frends.GoogleDrive.Download.Test/UnitTests.cs (hunk 3 above)
Comment (jefim) [reply Frends.GoogleDrive.Download/Frends.GoogleDrive.Download.Test/UnitTests.cs:39]: Answered in DM

**Review by MatteoDelOmbra — APPROVED** [review]

**Pull request merged**